### PR TITLE
Built in self test serial console

### PR DESCRIPTION
**Pull request by RyanMah123**

### CODE REVIEW

**Comment by RyanMah123** [comment]
I know there are a shit ton of files changed, but I tested this on hw and it works. The changed files are because I had to change the IOC and regenerate the libraries.

**Comment by RyanMah123** [comment]
Also, we need to talk about all the stuff I had to fix eventually...

File: STM32F405RGTx/Core/Src/rtos_threads/state_machine.cpp [hunk 1]
```diff
@@ -518,7 +518,7 @@ void StateMachineThread::runStateMachine(void *argument) {
             Error_Handler();
     }
 	// CurrentState = (*StateMachineThread::SM[CurrentState].Event)();
-    sendCANHeartbeat();
+    // sendCANHeartbeat();
```
Comment (GuyBlumenthal) [inline STM32F405RGTx/Core/Src/rtos_threads/state_machine.cpp]: What's the reason you removed the can heartbeat

**Review by GuyBlumenthal — APPROVED** [review]

**Comment by GuyBlumenthal** [comment]
> Also, we need to talk about all the stuff I had to fix eventually...

What do you mean exactly? The ioc stuff or the general code stuff

File: STM32F405RGTx/Core/Src/rtos_threads/state_machine.cpp (hunk 1 above)
Comment (RyanMah123) [reply STM32F405RGTx/Core/Src/rtos_threads/state_machine.cpp]: debugging thing I forgot to uncomment, I'll fix before merging

**Comment by RyanMah123** [comment]
> > Also, we need to talk about all the stuff I had to fix eventually...
> 
> What do you mean exactly? The ioc stuff or the general code stuff

Both, I'm less mad about it now, but I did spend ~4 hours last night on it.

**Pull request merged**